packages/graphql/src/schema/validation/custom-rules/directives/relationship.ts

```diff
@@ -149,3 +158,23 @@ export function validateRelationshipDirective(context: Neo4jValidationContext):
         },
     };
 }
+
+function validateRelationshipTarget(
+    fieldDefinitionNode: FieldDefinitionNode,
+    typeMapWithExtensions: TypeMapWithExtensions
+): void {
+    const relatedTypename = getInnerTypeName(fieldDefinitionNode.type);
+    const relatedType = typeMapWithExtensions[relatedTypename]?.definition;
+    if (!relatedType) {
+        return;
+    }
+
+    if (relatedType.kind === Kind.OBJECT_TYPE_DEFINITION) {
```

Comment:
Interfaces & Union will be address in this PR or in a follow-up PR?

Reply:
This will be covered in a separate PR, as it will use the changes in #6159